lib/rory/request_parameter_logger.rb

```diff
@@ -0,0 +1,58 @@
+require_relative 'parameter_filter'
+
+module Rory
+  class RequestParameterLogger
+
+    def initialize(app, logger=nil, filters=[:password, :tin, :ssn, :social_security_number, :file_attachment])
+      @app = app
+      @logger = logger
+      @filters = filters
+    end
+
+    def call(env)
+      @env = env
+      @env['rack.input'].rewind
+      log_request
+      @app.call(@env)
+    end
+
+    private
+
+    def log_request
+      log_message(request_signature)
+      log_message("Parameters: #{filtered_params}")
+    end
+
+    def logger
+      @logger || @env['rack.errors']
+    end
+
+    def log_message(message)
+      if logger.respond_to?(:write)
+        logger.write(message + "\n")
+      else
+        logger.info(message)
```

Comment:
Look at what `Rack::Commonlogger` does.. I'd argue for doing the same ourselves:

https://github.com/rack/rack/blob/master/lib/rack/commonlogger.rb#L58